rpc/dialer.go

```diff
@@ -290,17 +290,13 @@ func dialDirectGRPC(ctx context.Context, address string, dOpts dialOptions, logg
 	if dOpts.unaryInterceptor != nil {
 		unaryInterceptors = append(unaryInterceptors, dOpts.unaryInterceptor)
 	}
-	unaryInterceptor := grpc_middleware.ChainUnaryClient(unaryInterceptors...)
-	dialOpts = append(dialOpts, grpc.WithUnaryInterceptor(unaryInterceptor))
 
 	var streamInterceptors []grpc.StreamClientInterceptor
 	streamInterceptors = append(streamInterceptors, grpc_zap.StreamClientInterceptor(grpcLogger))
 	streamInterceptors = append(streamInterceptors, StreamClientTracingInterceptor())
 	if dOpts.streamInterceptor != nil {
 		streamInterceptors = append(streamInterceptors, dOpts.streamInterceptor)
 	}
-	streamInterceptor := grpc_middleware.ChainStreamClient(streamInterceptors...)
```

Comment:
I also moved `streamInterceptor`s creation to below the auth logic to not weirdly have stream interceptor creation here and unary interceptor creation somewhere else.